### DISCUSSION TITLE
Don't delay subsequent page visits again, but only for first x minutes. OR: Option to cancel current override when tab is closed or on site outside block set

### DISCUSSION THREAD

**Discussion by xuiqzy** [proginosko/LeechBlockNG#399] [Ideas]
When accessing a paged from a block set which has a delayed page, I wait the delay and then continue browsing in that tab or opening new links directly from that tab. Opening a new tab and manually going to a URL from that block set shows the delayed page again in contrast (which is also not an intuitive different behavior).

The result I'm looking after is: Being reminded after a while to consider again (with a delay I have to wait) if I want to continue continuously browsing sites from the block set and not being lost in it after waiting out the delay once. But still seeing the delay if I newly decide to visit a site from the block set every time I newly open one.

It could also be similar to "Attempt to reload pages every x seconds (only applies to default page)" option.
Either it could be just that, but also work for block sets that use the delayed page. This may be the easiest to implement and enough for the basic usage (I don't usually am mid text edit on my blocked sites, but it would make that a problem, but also force people off the site mid visit that need that).

Or, better, it would only be triggered by clicking something that results in a new URL being set / being navigated to in the tab and then showing the delayed page again. This way, it would not abruptly abort what I am watching/reading/editing on the current page (not needed to be snatched out of the current site visit immediately for my use case, might be different if someone is watching an hour-long YouTube video and wants to be pulled from that after the timeout.)
So, it would be like having an exception for the "Block only first accessed page of site when delaying page is used" feature that says: "*Do* block the next site visit again after x minutes have passed since the first delay page" or "Don't block subsequent page visits again for the first x minutes". This would be the most intuitive solution I think.

Optimally, and this would also fix the losing some editing on the site issue, would be a full page overlay of the delay page instead of a redirect to the delay page. And use that again as soon as the window of e.g. 30 minutes has passed. After the delay, it would then remove the overlay and return to the site with all its current state intact. This is probably way too complicated to implement easily, but just leaving the idea here. Maybe more suited to fixing the mid-editing issue.

What might be easier to accomplish altogether for the same result would be to use the temporary override for a blocked site for a specific amount of time. But that initial and subsequent temporary override of the block would then need a countdown delay option similar to the delay page to achieve the same delay effect on the first and any subsequent overrides. I'll try using the access code of sufficient length to simulate the delay for now. See [Idea: Different override ''tasks''](https://github.com/proginosko/LeechBlockNG/discussions/371).
Problem with that: I actually appreciate seeing the delay again if I only visit a blocked site for 10 minutes and then actively decided to close it and then open another blocked site 10 minutes later again. So not staying on the page for 30 minutes, where I don't want a delay when visiting a new page in the block set from a current page in the block set. For the temporary override solution, that close and then open case does not cause any block/delay with e.g. a 30-minute temporary override. I could try setting the temporary override duration short enough, so it asks me again for this close and open again case but still not be annoying enough when I want to stay on the site as a workaround.
It would need to cancel the current temporary override as soon as the current tab (or rather all tabs created since override visiting a block set page) visits a page outside the block set or is closed. This would lead to a new temporary override needing to be activated by the user on the next visit, even when the next visit is only a few minutes later.
So, the thing I want is being able to spend e.g. 30 minutes on the site on different pages and then see a delay again, but if I close the tab / visit a site outside the block set and open the blocked site newly, I want to see the delay immediately every time.

Sorry for the rambling!

This is just an idea, I know a solution might be difficult, but maybe there is a simpler solution that I wasn't thinking of.

**Comment by Armnoo** [comment]
Hey, I wonder if you have a workaround for this feature. I want to delay 1.5 hours after 30 minute use.

**Reply by proginosko** [reply]
I plan to add an option to specify a minimum duration for the block (e.g., "Always block sites for at least __ minutes"). That should provide the functionality you want.
